alacritty_terminal/src/ansi.rs

```diff
@@ -1392,6 +1394,17 @@ fn attrs_from_sgr_parameters(params: &mut ParamsIter<'_>) -> Vec<Option<Attr>> {
                 parse_sgr_color(&mut iter).map(Attr::Background)
             },
             [49] => Some(Attr::Background(Color::Named(NamedColor::Background))),
+            [58] => {
+                let mut iter = params.map(|param| param[0]);
+                parse_sgr_color(&mut iter).map(|color| Attr::UnderlineColor(Some(color)))
+            },
+            [58, params @ ..] => {
+                let rgb_start = if params.len() > 4 { 2 } else { 1 };
+                let rgb_iter = params[rgb_start..].iter().copied();
+                let mut iter = iter::once(params[0]).chain(rgb_iter);
```

Comment:
This is copy/pasted from the 38/48 escape I'd assume. At this point it might be worth it to extract it to a function? Especially because it's not 100% obvious what this does and a comment about the format might actually help readers.